src/Aspire.Hosting/ApplicationModel/ResourceCommandAnnotation.cs

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+/// <summary>
+/// Represents a command annotation for a resource.
+/// </summary>
+[DebuggerDisplay("Type = {GetType().Name,nq}, Source = {Source}, Target = {Target}")]
+public sealed class ResourceCommandAnnotation : IResourceAnnotation
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ResourceCommandAnnotation"/> class.
+    /// </summary>
+    public ResourceCommandAnnotation(string type, string displayName, Func<CustomResourceSnapshot, Task<bool>> visible, Func<string, IServiceProvider, Task> invokeCommand, string? iconContent, bool isHighlighted)
+    {
+        ArgumentNullException.ThrowIfNull(type);
+        ArgumentNullException.ThrowIfNull(displayName);
+
+        Type = type;
+        DisplayName = displayName;
+        Visible = visible;
+        InvokeCommand = invokeCommand;
+        IconContent = iconContent;
+        IsHighlighted = isHighlighted;
```

Comment:
What does it look like when the command is "highlighted" vs "not highlighted"? How does it affect command operation?

The "highlighted" name suggests "UI styling" which makes me wonder if this information should be part of the view model...

Reply:
See https://github.com/dotnet/aspire/pull/5516. The difference is whether the command is directly displayed on the resources page or is nested in the actions menu.

> The "highlighted" name suggests "UI styling" which makes me wonder if this information should be part of the view model...

Eventually this flag is used in the UI, but the info for whether a command is highlighted or not needs to come from somewhere. That's why it's part of the annotation (along with other display info such as display name and icon).

Reply:
Got it.